fractions/microprofile/microprofile-openapi/src/main/java/org/wildfly/swarm/microprofile/openapi/runtime/OpenApiHttpHandler.java

```diff
@@ -39,31 +40,21 @@
     private static final String OAI = "/openapi";
     private static final String ALLOWED_METHODS = "GET, HEAD, OPTIONS";
     private static final String QUERY_PARAM_FORMAT = "format";
-    private static final String OAI_JSON;
-    private static final String OAI_YAML;
-
-    static {
-        try {
-            OpenAPI model = OpenApiDocument.INSTANCE.get();
-            OAI_JSON = OpenApiSerializer.serialize(model, Format.JSON);
-            OAI_YAML = OpenApiSerializer.serialize(model, Format.YAML);
-        } catch (IOException ioe) {
-            throw new RuntimeException(ioe);
-        }
-    }
 
+    private final Map<Format, String> cachedModels;
     private final HttpHandler next;
 
     public OpenApiHttpHandler(HttpHandler next) {
         this.next = next;
+        this.cachedModels = new ConcurrentHashMap<>();
     }
 
     /**
      * @see io.undertow.server.HttpHandler#handleRequest(io.undertow.server.HttpServerExchange)
      */
     @Override
     public void handleRequest(HttpServerExchange exchange) throws Exception {
-        if (OAI.equalsIgnoreCase(exchange.getRequestPath())) {
+        if (OAI.equalsIgnoreCase(exchange.getRequestPath()) && OpenApiDocument.INSTANCE.isSet()) {
```

Comment:
Is there a situation that could occur where the model wouldn't be set on `OpenApiDocument` and this is called?

Reply:
Yes. This issue is all about graceful startup, i.e. processing http requests *before* the app is deployed.

Reply:
Is it about processing http requests, or just that Undertow initialized the class?

I would expect that Undertow initializing the class, and thus the static method being called, doesn't mean it's handling requests, merely that Undertow is being setup ready to take requests.

That's why I wasn't understanding at the point we're accepting a request, that the model wouldn't be set on `OpenApiDocument`?

Reply:
> Is it about processing http requests, or just that Undertow initialized the class?

I have to find out.

Reply:
@kenfinnigan You were right, it seems the handler is not used until the app is deployed. I will remove the needless condition.

Reply:
Ok, in that case, should a message be logged that it's not being processed for that reason?

Otherwise it just passes through and does nothing when it was expected it would

Reply:
The client receives HTTP status 404 (until the app is deployed) which is imho OK.